streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -51,10 +52,16 @@ public SourceNode(final String name,
     }
 
     KIn deserializeKey(final String topic, final Headers headers, final byte[] data) {
+        if (keyDeserializer == null) {
```

Comment:
In which case can `keyDeserializer` be `null` here?

Should we not better ensure that it's never `null` to begin with? Have the check in `init()` method?

Reply:
Ahh good catch, I did a check in `WrappingNullableUtils` for the serializer but missed it for the deserializers somehow, I added it there and removed this. Now we throw the config error when we try to build a topology without a default or a serde passed in